doc/src/redis.rst

```diff
@@ -15,23 +15,57 @@ Configuration
 -------------
 
 Out of the box, Redis is set up with a couple of common default
-parameters and listens on *localhost* and the IP-addresses of the
-*ethsrv*-interface of your VM (See :ref:`networking` for details on this topic).
-
-listens on the *ethsrv* interface on port 6379.
-
-If you need to change the behaviour of Redis, you can put your
-additional configuration into :file:`/etc/local/redis/custom.conf`.
-
-Available configuration options can be found in the
-`official documentation <https://redis.io/topics/config>`_.
+parameters and listens on the IP-addresses of the *loopback* (localhost) and
+*ethsrv*-interfaces of your VM on port 6379 (See :ref:`networking`
+for details on this topic).
+
+If you need to change the behaviour of Redis, you define your redis
+configuration with the NixOS options ``flyingcircus.services.redis.*``.
+
+See :ref:`nixos-custom-modules` for general information about writing custom NixOS
+modules in :file:`/etc/local/nixos`.
+
+The following NixOS module sets the listening addresses to ``203.0.113.54`` and
+``203.0.113.57`` as well as overriding the password to ``foobarpass``. The maximum
+memory size is set to ``512mb``. The exact behavior Redis follows when the maxmemory
+limit is reached is configured using the ``maxmemory-policy`` configuration directive
+and is set to ``noeviction`` in this example. Read more at `redis topic lru cache <https://redis.io/topics/lru-cache>`.
+
+.. code-block:: nix
+
+    # /etc/local/nixos/redis.nix
+    { ... }:
+    {
+        flyingcircus.services.redis = {
+            listenAddresses = [ "203.0.113.54", "203.0.113.57 "];
+            password = "foobarpass";
+            maxMemory = "512mb";
+            maxMemoryPolicy = "noeviction";
+        };
+    }
+
+As an alternative to setting the ``maxMemory`` by hand you can set a ``memoryPercentage``
+option. This will set the memory limit to a percentage of the total memory of the
+system.
+
+.. code-block:: nix
+
+    # /etc/local/nixos/redis.nix
+    { ... }:
+    {
+        flyingcircus.services.redis = {
+            memoryPercentage = "50";
+        };
+    }
 
 For further information on how to activate changes on our NixOS-environment,
 please consult :ref:`nixos-local`.
 
 The authentication password is automatically generated upon installation
 and can be read *and changed* by service users. It can be found in
-:file:`/etc/local/redis/password`.
+:file:`/etc/local/redis/password`. It can also be specified in the
+``flyingcircus.services.redis.password`` option where the password
+will have a higher priority than the one in the filesystem.
```

Comment:
add a notice that using the option will add the password as world-readable to the nix store.